File: ALDS1_1_B.py
# -*- coding:utf-8 -*-
def gcd(n,m):
    if n < m:
        tmp = m
        m = n
        n = tmp
    while m > 0:
        r = n % m
        n = m
        m = r
    return n
    
x, y = map(int,input().split())
print(gcd(x,y))

File: ALDS1_1_B.cpp
#include<iostream>
#include<cmath>
using namespace std;

int gcd(int a, int b) {
  int tmp;
  if (a < b) {
    tmp = b;
    b = a;
    a = tmp;
  }

  int c;
  
  while (b != 0) {
    c = a % b;
    a = b;
    b = c;
  }

  return a;
  
}

int main() {

  int x, y;
  cin >> x >> y;

  cout << gcd(x, y) << endl;
  
  return 0;
}
